src/storage/kv/entry.rs

```diff
@@ -31,6 +32,7 @@ impl Entry {
             metadata: None,
             value: Bytes::copy_from_slice(value),
             ts: 0,
+            savepoint_no: 0,
```

Comment:
Can we please include this in the write_set inside a transaction, making it a Vec<{Entry, savepoint_no}>?

The reason is that an entry is used for translation of records across multiple places (store, compaction etc) and this savepoint no is a property which is just relevant to the transaction internals. It will help avoid confusion later on in the entry.rs internals

Reply:
Sure, I actually implemented it as a vec of tuples first, but then decided to embed the savepoint no into Entry to save little bit on allocations when committing. But yeah, I guess in this case a small win in perf is not really worth it.

Reply:
Thank you!